packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -0,0 +1,222 @@
+    private readonly afterHandlers: Set<EditableTreeContextHandler> = new Set();
+    private readonly nodes: Map<ETreeNodePath, ProxyTarget | ProxyTargetSequence> = new Map();
+    private emptyNode?: ProxyTarget;
+
+    constructor(
+        public readonly forest: IEditableForest,
+        public readonly tree?: ISharedTree,
+    ) {
+        const observer = new SimpleObservingDependent((token?: InvalidationToken, delta?: Delta.Root): void => {
+            if (token?.isSecondaryInvalidation) {
+                // TODO: this is only an example of after change processing, which is not supported yet by ObjectForest/SharedTree.
+                this.handleAfterChange();
+            } else {
+                this.prepareForEdit();
+            }
+        });
+        this.observers.push(observer);
+        forest.registerDependent(observer);
+    }
+
+    public prepareForEdit(): void {
+        for (const target of this.withCursors) {
+            target.prepareForEdit();
+        }
+        assert(this.withCursors.size === 0, 0x3c0 /* prepareForEdit should remove all cursors */);
+    }
+
+    public free(): void {
+        for (const target of this.withCursors) {
+            target.free();
+        }
+        for (const target of this.withAnchors) {
+            target.free();
+        }
+        for (const observer of this.observers) {
+            this.forest.removeDependent(observer);
+        }
+        assert(this.withCursors.size === 0, 0x3c1 /* free should remove all cursors */);
+        assert(this.withAnchors.size === 0, 0x3c2 /* free should remove all anchors */);
+    }
+
+    public get root(): UnwrappedEditableField {
+        const cursor = this.forest.allocateCursor();
+        const destination = this.forest.root(this.forest.rootField);
+        const cursorResult = this.forest.tryMoveCursorTo(destination, cursor);
+        const targets: ProxyTarget[] = [];
+        if (cursorResult === TreeNavigationResult.Ok) {
+            this.emptyNode?.free();
+            this.emptyNode = undefined;
+            do {
+                targets.push(this.createTarget(cursor));
+            } while (cursor.seek(1) === TreeNavigationResult.Ok);
+        }
+        cursor.free();
+        this.forest.anchors.forget(destination);
+        const rootSchema = lookupGlobalFieldSchema(this.forest.schema, rootFieldKey);
+        const fieldKind = getFieldKind(rootSchema);
+        if (targets.length === 0 && fieldKind.multiplicity !== Multiplicity.Sequence) {
+            targets.push(this.createEmptyTarget());
+        }
+        return proxifyField(this, getFieldKind(rootSchema), targets);
+    }
+
+    public createEmptyTarget(): ProxyTarget {
+        if (!this.emptyNode) {
+            this.emptyNode = new ProxyTarget(this);
+        }
+        return this.emptyNode;
+    }
+
+    public createTarget(cursor: ITreeSubscriptionCursor): ProxyTarget {
+        const path = (cursor as unknown as RootedTextCursor).getPath();
```

Comment:
This is very much not a safe thing to do: This is assuming all cursors here use a particular implementation.

At the very least this should note that this is not ok, and that we should fix it.

```suggestion
        // TODO: remove this assumption about the underlying cursor type once migrated to the new cursor API
        // This happens to work for use with ObjectForest but is not in general safe.
        const path = (cursor as unknown as RootedTextCursor).getPath();
```